spring-cloud-bus/src/main/java/org/springframework/cloud/bus/BusAutoConfiguration.java

```diff
@@ -182,8 +183,8 @@ public PathMatcher busPathMatcher() {
 
 		@Bean
 		public ServiceMatcher serviceMatcher(@BusPathMatcher PathMatcher pathMatcher,
-				BusProperties properties) {
-			ServiceMatcher serviceMatcher = new ServiceMatcher(pathMatcher, properties.getId());
+				BusProperties properties, @Value("${spring.cloud.config.name:}") String[] configNames) {
```

Comment:
Use `Environment` rather than `@Value`, otherwise relaxed binding won't work.

Reply:
Thanks for that feedback. I was aware of the relaxed binding differences between Spring Boot and Spring Core, but as `Environment` also is Spring Core, i expected them to behave in the same way, as described here: https://stackoverflow.com/questions/26897314/differences-between-value-annotation-and-the-environment-api and as suggested by the `Environment` class javadoc. 

Would you mind and give an example of something that works in `Environment` which does not work in `Value`? This is just out of curiousity, i am happy to do the change, i was not really happy with using `Value`, but could not pinpoint it.

Reply:
Relaxed binding is a feature of spring boot, not core Framework.

Reply:
Yes, that is why i do not get you comment completely. Using `@Value` will make relaxed binding not work, but using `Environment` will make it work? I am fine with it, but do not understand that difference, sorry. Still trying to find something through google, but all i find says "use Environment when you care about profile information, else use @Value". Nowhere a hint on a difference in regard to relaxed binding.

Reply:
Updated the PR to use `Environment` over `@Value`.